CollAction/Services/Donation/DonationService.cs

```diff
@@ -184,6 +269,81 @@ public async Task Charge(string sourceId)
             }
         }
 
+        public async Task<IEnumerable<Subscription>> GetSubscriptionsFor(ApplicationUser userFor)
+        {
+            var customers = await _customerService.ListAsync(new CustomerListOptions()
+            {
+                Email = userFor.Email
+            });
+
+            var subscriptions = await Task.WhenAll(
+                customers.Select(c =>
+                    _subscriptionService.ListAsync(new SubscriptionListOptions()
+                    {
+                        CustomerId = c.Id
+                    })));
+
+            return subscriptions.SelectMany(s => s);
+        }
+
+        public async Task CancelSubscription(string subscriptionId, ApplicationUser userFor)
+        {
+            Subscription subscription = await _subscriptionService.GetAsync(subscriptionId);
+            Customer customer = await _customerService.GetAsync(subscription.CustomerId);
+
+            if (!customer.Email.Equals(userFor.Email, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidOperationException($"User {userFor.Email} doesn't match subscription e-mail {subscription.Customer.Email}");
+            }
+
+            subscription = await _subscriptionService.CancelAsync(subscriptionId, new SubscriptionCancelOptions());
+
+            _context.DonationEventLog.Add(new DonationEventLog()
+            {
+                UserId = userFor.Id,
+                Type = DonationEventType.Internal,
+                EventData = subscription.ToJson()
+            });
+            await _context.SaveChangesAsync();
+        }
+
+        private async Task<Plan> CreateRecurringPlan(int amount, string currency)
+        {
+            Product product = await GetOrCreateRecurringDonationProduct();
+            return await _planService.CreateAsync(new PlanCreateOptions()
+            {
+                ProductId = product.Id,
+                Active = true,
+                Amount = amount * 100,
+                Currency = currency,
+                Interval = "month",
+                BillingScheme = "per_unit",
+                UsageType = "licensed",
+                IntervalCount = 1
+            });
+        }
+
+        private async Task<Product> GetOrCreateRecurringDonationProduct()
+        {
+            var products = await _productService.ListAsync(new ProductListOptions()
+            {
+                Active = true,
+                Type = "service"
+            });
+            Product product = products.FirstOrDefault(p => p.Name == RecurringDonationProduct);
```

Comment:
No var?

Reply:
I only use "var" if the type is completely obvious, or if the full type is too long (Dictionary<...> etc). Otherwise I find it better to be a little explicit.